// react-native-dbb-rtmp/LivePlayer.js
import React, { Component } from "react";
import { PropTypes } from "prop-types";
import {
  requireNativeComponent,
  View,
  UIManager,
  findNodeHandle
} from "react-native";

var RCT_LIVE_PLAYER_REF = "LivePlayer";

export default class LivePlayer extends Component {
  constructor(props) {
    super(props);
  }

  _onChange(event) {
    if (event.nativeEvent && event.nativeEvent.type === "status") {
      switch (event.nativeEvent.code) {
        //connecting
        case 1000:
          console.log("链接视频");
          break;
        //connected
        case 1001:
          console.log("链接成功");
          break;
        case 1002:
          console.log("链接失败");
          break;
        case 1003:
          console.log("重新链接");
          break;
        //play end
        case 1004:
          this.props.onEnd && this.props.onEnd();
          break;
        case 1005:
          console.log("视频播放中网络异常, 会进行自动重连");
          break;
        //cache loading
        case 1101:
          this.props.onLoading && this.props.onLoading();
          break;
        //cache loaded
        case 1102:
          this.props.onLoad && this.props.onLoad({ duration: 0 });
          break;
        default: {
        }
      }
    }
  }

  seek = percent => {};

  render() {
    return (
      <RCTLivePlayer
        {...this.props}
        srcUrl={
          this.props.source && this.props.source.uri
            ? this.props.source.uri
            : ""
        }
        ref={RCT_LIVE_PLAYER_REF}
        onChange={event => {
          this._onChange(event);
        }}
      />
    );
  }
}

LivePlayer.name = RCT_LIVE_PLAYER_REF;
LivePlayer.propTypes = {
  source: PropTypes.object,
  bufferTime: PropTypes.number,
  maxBufferTime: PropTypes.number,
  paused: PropTypes.bool,
  muted: PropTypes.bool,
  resizeMode: PropTypes.oneOf(["contain", "cover", "stretch"]),
  renderType: PropTypes.oneOf(["SURFACEVIEW", "TEXTUREVIEW"]),
  onLoading: PropTypes.func,
  onLoad: PropTypes.func,
  onEnd: PropTypes.func,
  ...View.propTypes
};

const RCTLivePlayer = requireNativeComponent("RCTLivePlayer", LivePlayer, {
  nativeOnly: { srcUrl: true, onChange: true }
});
